src/client/dfuse/pil4dfs/hook.c

```diff
@@ -483,39 +570,44 @@ get_module_maps(void)
 		}
 		iPos_Save = iPos;
 		/* find the next line */
-		iPos = get_position_of_next_line(szBuf, iPos + 38, read_size);
+		iPos = get_position_of_next_line(buf, iPos + 38, read_size);
 		if ((iPos - iPos_Save) > 73) {
 			/* with a lib name */
-			ReadItem = sscanf(szBuf + iPos_Save + 73, "%s", szLibName);
+			ReadItem = sscanf(buf + iPos_Save + 73, "%s", lib_name);
 			if (ReadItem == 1) {
-				if (strncmp(szLibName, "[stack]", 7) == 0) {
+				if (strncmp(lib_name, "[stack]", 7) == 0) {
 					num_seg--;
 					break;
 				}
-				if (query_lib_name_in_list(szLibName) == -1) {
+				if (query_lib_name_in_list(lib_name) == -1) {
 					/* a new name not in list */
-					lib_name_list[num_lib_in_map] = strdup(szLibName);
+					D_STRNDUP(lib_name_list[num_lib_in_map], lib_name,
+						  PATH_MAX);
 					if (lib_name_list[num_lib_in_map] == NULL) {
-						printf("Error> strdup(szLibName) failed.\nQuit\n");
+						D_FREE(buf);
+						D_FREE(lib_name);
+						D_FATAL("Failed to allocate memory for "
+							"lib_name_list[%d] %d (%s)\n",
+							num_lib_in_map, ENOMEM, strerror(ENOMEM));
 						exit(1);
 					}
 					lib_base_addr[num_lib_in_map] = addr_B;
 					num_lib_in_map++;
 					if (num_lib_in_map >= MAX_NUM_LIB) {
-						printf("Warning> lib_base_addr is FULL.\n"
+						D_WARN("lib_base_addr is FULL.\n"
```

Comment:
You shouldn't use \n in the middle of error messages, either reflow this comment or make it two calls to D_WARN.